pkg/view/component/signpost.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (c) 2021 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package component
+
+import (
+	"github.com/vmware-tanzu/octant/internal/util/json"
+)
+
+// Signpost is a component for a signpost
+// +octant:component
+type Signpost struct {
+	Base
+	Config SignpostConfig `json:"config"`
+}
+
+// SignpostConfig is the contents of a signpost
+type SignpostConfig struct {
+	// Component is the component that will be displayed.
+	Component Component `json:"component,omitempty"`
+	// Trigger is the component that will trigger the signpost.
+	Trigger Component `json:"trigger"`
+	// Message is the text that will be displayed.
+	Message string `json:"message"`
+	// Reverse is the order of the components.
+	Reverse bool `json:"reverse"`
+	// Column is the layout.
+	Column bool `json:"column"`
+	// Position of Signpost
+	Position Position `json:"position"`
+}
+
+type Position string
+
+const (
+	PositionTopLeft      Position = "top-left"
+	PositionTopMiddle    Position = "top-middle"
+	PositionTopRight     Position = "top-right"
+	PositionRightTop     Position = "right-top"
+	PositionRightMiddle  Position = "right-middle"
+	PositionRightBottom  Position = "right-bottom"
+	PositionBottomRight  Position = "bottom-right"
+	PositionBottomMiddle Position = "bottom-middle"
+	PositionBottomLeft   Position = "bottom-left"
+	PositionLeftBottom   Position = "left-bottom"
+	PositionLeftMiddle   Position = "left-middle"
+	PositionLeftTop      Position = "left-top"
+)
+
+func NewSignpost(c Component, t Component, m string) *Signpost {
+	so := &Signpost{
+		Base: newBase(TypeSignpost, nil),
+		Config: SignpostConfig{
+			Component: c,
+			Trigger:   t,
+			Message:   m,
+			Reverse:   false,
+			Column:    false,
+		},
+	}
+
+	return so
+}
+
+type signpostMarshal Signpost
+
+// MarshalJSON implements json.Marshaler
+func (t *Signpost) MarshalJSON() ([]byte, error) {
+	m := signpostMarshal(*t)
+	m.Metadata.Type = TypeSignpost
+	return json.Marshal(&m)
+}
+
+func (c *SignpostConfig) UnmarshalJSON(data []byte) error {
+	x := struct {
+		Component TypedObject `json:"component,omitempty"`
```

Comment:
This needs to be a pointer otherwise it won't be empty if nil

Reply:
Done!